### PR TITLE
WIP: Positions round 2

### PR DESCRIPTION
**Pull request by ultrasecreth**

### CODE REVIEW

**Review by egillh210 — APPROVED** [review]

File: packages/hardhat/contracts/Future.sol [hunk 1]
```diff
@@ -50,23 +44,24 @@ contract Future is IUniswapV3SwapCallback {
         }
     }
 
-    function long(uint quantity, uint price) external returns (uint amountReceived) {
+    function long(int quantity, uint price) external override returns (int amountReceived, int amountPaid) {
```
Comment (scelestino) [inline packages/hardhat/contracts/Future.sol:47]: why did you change from uint to int? can this number be negative? should we check these parameters?

**Review by scelestino — APPROVED** [review]

File: packages/hardhat/contracts/Future.sol (hunk 1 above)
Comment (ultrasecreth) [reply packages/hardhat/contracts/Future.sol:47]: I think it was to avoid some casting, I have a full review of the types and params in my todo list
